File: 56 Merge Intervals/Merge Intervals.java
import java.util.*;

class Solution {
    public int[][] merge(int[][] intervals) {
        Arrays.sort(intervals, (a, b) -> Integer.compare(a[0], b[0]));
        List<int[]> result = new ArrayList<>();
        result.add(intervals[0]);

        for (int i = 1; i < intervals.length; i++) {
            int[] current = intervals[i];
            int[] last = result.get(result.size() - 1);
            int lastEnd = last[1];

            if (current[0] <= lastEnd) {
                last[1] = Math.max(lastEnd, current[1]);
            } else {
                result.add(current);
            }
        }

        return result.toArray(new int[result.size()][]);
    }
}

File: 56 Merge Intervals/Merge Intervals.js
function merge(intervals) {
  intervals.sort((a, b) => a[0] - b[0]);
  let result = [intervals[0]];

  for (let i = 1; i < intervals.length; i++) {
    let start = intervals[i][0];
    let end = intervals[i][1];
    let lastEnd = result[result.length - 1][1];
    if (start <= lastEnd) {
      result[result.length - 1] = [
        result[result.length - 1][0],
        Math.max(lastEnd, end),
      ];
    } else {
      result.push([start, end]);
    }
  }
  return result;
}
